### PR TITLE
DUPLO-15935 fixed k8 secret diff issue

### PR DESCRIPTION
**Pull request by Duplo-nikv**
## Overview

Fixed k8 secret diff issue
## Summary of changes
Fixed k8 secret diff issue
This PR does the following:

- fixed DiffSuppressFunc function for secret_data attribute
- ...

## Testing performed

- [ ] Using unit tests
- [ ] Manually, on my local system
- [x] Manually, on a remote test system

## Describe any breaking changes

- ...

### CODE REVIEW

**Comment by zafarabbas** [comment]
Task linked: [DUPLO-15935 TF: duplocloud_k8_secret ignores manual changes to secret value](https://app.clickup.com/t/8655600/DUPLO-15935)

File: .terraform.tfstate.lock.info [hunk 1]
```diff
@@ -0,0 +1 @@
+{"ID":"7938f8a6-be9d-140d-bdc7-bc4f779a662b","Operation":"OperationTypeApply","Info":"","Who":"nikhil@TGM9LHHH7Y-nikhil","Version":"1.7.3","Created":"2024-03-26T05:50:07.513638Z","Path":"terraform.tfstate"}
```
Comment (tahir-duplo) [inline .terraform.tfstate.lock.info:1]: Remove this file

Comment (Duplo-nikv) [reply .terraform.tfstate.lock.info]: done

File: duplocloud/resource_duplo_k8_secret.go [hunk 2]
```diff
@@ -198,13 +198,7 @@ func resourceK8SecretDelete(ctx context.Context, d *schema.ResourceData, m inter
 }
 
 func diffIgnoreForSecretMap(k, old, new string, d *schema.ResourceData) bool {
```
Comment (tahir-duplo) [inline duplocloud/resource_duplo_k8_secret.go:200]: Do we need this function if we are using  old == new

Comment (Duplo-nikv) [reply duplocloud/resource_duplo_k8_secret.go:200]: Not needed ,will remove it

**Review by tahir-duplo — APPROVED** [review]
LGTM

**Pull request merged**